backend/routes.js

```diff
@@ -41,23 +41,29 @@ route.POST('/event', {
 }, async function (request, h) {
   try {
     console.debug('/event handler')
-    const entry = GIMessage.deserialize(request.payload)
+    const deserializedHEAD = GIMessage.deserializeHEAD(request.payload)
     try {
-      await sbp('backend/server/handleEntry', entry)
+      await sbp('backend/server/handleEntry', deserializedHEAD, request.payload)
     } catch (err) {
-      if (err.name === 'ChelErrorDBBadPreviousHEAD') {
-        console.error(err, chalk.bold.yellow('ChelErrorDBBadPreviousHEAD'))
-        const HEADinfo = await sbp('chelonia/db/latestHEADinfo', entry.contractID()) ?? { HEAD: null, height: 0 }
+      if (err.name === 'ChelErrorDBBadPreviousHEAD' || err.name === 'ChelErrorAlreadyProcessed') {
+        console.error(err, chalk.bold.yellow(err.name))
+        const HEADinfo = await sbp('chelonia/db/latestHEADinfo', deserializedHEAD.contractID) ?? { HEAD: null, height: 0 }
         const r = Boom.conflict(err.message, { HEADinfo })
         Object.assign(r.output.headers, {
           'shelter-headinfo-head': HEADinfo.HEAD,
           'shelter-headinfo-height': HEADinfo.height
         })
         return r
+      } else if (err.name === 'ChelErrorSignatureError') {
+        console.error(err, chalk.bold.yellow(err.name))
+        return Boom.badData('Invalid signature')
+      } else if (err.name === 'ChelErrorSignatureKeyUnauthorized') {
+        console.error(err, chalk.bold.yellow(err.name))
```

Comment:
Please DRY the repetition of these 3 `console.error(err, chalk.bold.yellow(err.name))` statements into a single one at the start of the `catch` block